pkg/snapshot/controllers/snapshotschedule.go

```diff
@@ -246,6 +264,45 @@ func (s *SnapshotScheduleController) updateVolumeSnapshotStatus(snapshotSchedule
 	return nil
 }
 
+// cleanupErroredSnapshots cleans any volumesnapshot CRs that are in Error state. It doesn't remove them
+// from the volumesnapshotschedule items list though.
+func (s *SnapshotScheduleController) cleanupErroredSnapshots(snapshotSchedule *stork_api.VolumeSnapshotSchedule) (err error) {
+	// Filter out the errored out snapshots that are older than the cleanup cutoff period.
+	for policy, policyVolumeSnapshot := range snapshotSchedule.Status.Items {
+		for idx, snapshot := range policyVolumeSnapshot {
+			if snapshot.Status == snapv1.VolumeSnapshotConditionError {
+				// Fetch the latest status as well in order to mitigate any late volumesnapshot updates.
+				snapshotStatus, err := getVolumeSnapshotStatus(snapshot.Name, snapshotSchedule.Namespace)
+				if err != nil {
+					log.VolumeSnapshotScheduleLog(snapshotSchedule).Warnf("Error getting status of snapshot %s: %s", snapshot.Name, err.Error())
+					continue
+				}
+
+				// Filter out the volumesnapshots that are older than the cutoff period and delete them.
+				snapshotCreationTime := snapshot.CreationTimestamp.Time
+				if snapshotStatus == snapv1.VolumeSnapshotConditionError && time.Since(snapshotCreationTime) > errorSnapshotCleanupCutoffTime {
+					log.VolumeSnapshotScheduleLog(snapshotSchedule).Infof("Going to delete the errored out snapshot: %v, age: %v", snapshot.Name, time.Since(snapshotCreationTime))
+					err := k8sextops.Instance().DeleteSnapshot(snapshot.Name, snapshotSchedule.Namespace)
+					if err != nil {
```

Comment:
Didn't get this. In this case, it will only go inside the `if` block if the snapshot is not found. What if in some case another error comes up while deleting snapshot?

Reply:
Following way I was referring.

`if err != nil && !k8s_errors.IsNotFound(err)` instead of `if err != nil`

Reply:
Done